### ISSUE TITLE
Magento 2 :-  Number of Lines in a Street Address not setting to default when you checked Use system value in Magento 2.1.7

### ISSUE THREAD

**Issue by pradeeprcs** [magento/magento2#13675]
### Preconditions

1. M 2.1.7

### Steps to reproduce

1.  Login to admin and got to Store->configuration->Customer->Customer Configuration-> Name and Address Options
2.  Set Number of Lines in a Street Address to 3 and save
3.  in front end it display 3 street line
4.  Next click on Use system value and save  it
5. Clear cache 

Still its show three line for street in address 

### Expected result

1.  it should display default street no that is two when you unset the value from Use system value

### Actual result

1. it display last saved values instead of default value (i'e 2 )

**Comment by magento-engcom-team** [comment]
@pradeeprcs, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by pradeeprcs** [comment]
i found issue  but no idea how to commit so i posting my solution below 

in Magento\Customer\Model\Config\Backend\Address\Street  replace afterDelete() with below code 


         public function afterDelete()
    {
        $result = parent::afterDelete();

        if ($this->getScope() == \Magento\Store\Model\ScopeInterface::SCOPE_WEBSITES) {
            $attribute = $this->_eavConfig->getAttribute('customer_address', 'street');
            $website = $this->_storeManager->getWebsite($this->getScopeCode());
            $attribute->setWebsite($website);
            $attribute->load($attribute->getId());
            $attribute->setData('scope_multiline_count', null);
            $attribute->save();
        }else{
			$attribute = $this->_eavConfig->getAttribute('customer_address', 'street');
            $attribute->setData('multiline_count', 2);
            $attribute->save();
		}

        return $result;
    }

**Comment by willtran** [comment]
I'm working on this #MLAU18

**Comment by tomekjordan** [comment]
Confirmed in any other Magento versions 2.2.6, 2.3.0 beta 32

**Comment by raulsinapsis** [comment]
I think problem is here **vendor/amzn/amazon-pay-module/Setup/UpgradeData.php**

```
    /**
     * @throws LocalizedException
     * @return void
     */
    private function upgradeAddressStreetMultiline(ModuleDataSetupInterface $setup)
    {
        $eavSetup = $this->eavSetupFactory->create(['setup' => $setup]);
        $row = $eavSetup->getAttribute('customer_address', 'street', 'multiline_count');

        if ($row === false || ! is_numeric($row)) {
            throw new LocalizedException(__('Could not find the "multiline_count" config of the "street" ' .
                'Customer address attribute.'));
        }

        if ($row < 3) {
            $eavSetup->updateAttribute('customer_address', 'street', 'multiline_count', 3);
        }
    }
```

**Comment by magento-engcom-team** [comment]
Hi @XxXgeoXxX. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @pradeeprcs. Thank you for your report. 
The issue has been fixed in magento/magento2#20565 by @XxXgeoXxX in 2.3-develop branch
Related commit(s):
 - [f3d410b0c23a868c319a9aa0c8a6d3eb3090c673](https://github.com/magento/magento2/commit/f3d410b0c23a868c319a9aa0c8a6d3eb3090c673)

The fix will be available with the upcoming 2.3.1 release.

**Comment by magento-engcom-team** [comment]
Hi @pradeeprcs. Thank you for your report. 
The issue has been fixed in magento/magento2#20566 by @XxXgeoXxX in 2.2-develop branch
Related commit(s):
 - [ca9bb265526f1461ee8647ed9de8ad81d13e36ce](https://github.com/magento/magento2/commit/ca9bb265526f1461ee8647ed9de8ad81d13e36ce)

The fix will be available with the upcoming 2.2.9 release.
